### ISSUE TITLE
stat_binhex() fails to notify users that they need to install hexbin package

### ISSUE THREAD

**Issue by yutannihilation** [tidyverse/ggplot2#4736]
https://github.com/tidyverse/ggplot2/pull/4671#issuecomment-1046090308

I can reproduce this on my local without hexbin installed.

```
> library(ggplot2)
> d <- ggplot(diamonds, aes(carat, price))
> d + geom_hex()
Warning message:
Computation failed in `stat_binhex()`:

```

I think this is because `check_installed` is inside `compute_group`, where the error is ignored.

https://github.com/tidyverse/ggplot2/blob/c89c265a57fd71f8a0288ce81037296aadc0a012/R/stat-binhex.r#L52

**Comment by yutannihilation** [comment]
Related: https://github.com/tidyverse/ggplot2/issues/4843

**Comment by yutannihilation** [comment]
Resolved by #4856
